projects/storefrontlib/src/shared/components/list-navigation/pagination/pagination.component.html

```diff
@@ -1,8 +1,67 @@
-<ngb-pagination
-  [collectionSize]="pagination.totalResults"
-  [page]="pagination.currentPage + 1"
-  (pageChange)="pageChange($event)"
-  [maxSize]="3"
-  [pageSize]="pagination.pageSize"
->
-</ngb-pagination>
+<!-- pagination -->
+<ul class="pagination">
+  <!-- Previous -->
+  <li class="page-item" [ngClass]="{ disabled: pagination.currentPage === 0 }">
+    <a class="page-link" (click)="clickPageNo(pagination.currentPage)">«</a>
+  </li>
+
+  <!-- First -->
+  <li class="page-item" [ngClass]="{ active: pagination.currentPage === 0 }">
+    <a class="page-link" (click)="clickPageNo(1)">{{ 1 }}</a>
+  </li>
+
+  <!-- Continued Backwards -->
+  <li class="page-item disabled" [hidden]="pagination.currentPage <= 2">
+    <a class="page-link">...</a>
+  </li>
+
+  <!-- Page Index -->
+  <li
+    class="page-item"
+    *ngFor="let page of Array(pagination.totalPages); let i = index"
+    [ngClass]="{ active: pagination.currentPage === i }"
+    [hidden]="
```

Comment:
I would move this complicated logic to `.ts` file
NOTE: do the same for each complicated logic in this file

Reply:
create some variables, like `isNextStepAvailable`, etc